File: spring-backend/src/main/java/com/petcare/repository/PetRepository.java
package com.petcare.repository;

/*
 * Comentario de modulo PetCare:
 * Repositorio legacy. Expone consultas de persistencia para entidades antiguas.
 */

import com.petcare.model.Pet;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface PetRepository extends JpaRepository<Pet, Integer> {
    List<Pet> findByOwnerId(Integer ownerId);
}

File: src/main/kotlin/com/petcare/repository/PetRepository.kt
package com.petcare.repository

/*
 * Comentario de modulo PetCare:
 * Repositorio de persistencia. Expone consultas a PostgreSQL mediante Spring Data.
 */

import com.petcare.model.Pet
import org.springframework.data.jpa.repository.JpaRepository

interface PetRepository : JpaRepository<Pet, Int> {
    fun findByOwnerIdOrderByCreatedAtDesc(ownerId: Int): List<Pet>
}
